apps/wistia-videos/src/components/Field.tsx

```diff
@@ -29,39 +31,51 @@ const StyledImageContainer = styled.div`
   }
 `;
 
+const VIDEOS_PER_PAGE = 100;
+
 const Field = (props: FieldProps) => {
   const { sdk } = props;
-  const [data, updateData] = useState<WistiaItem[] | []>([]);
+  const [data, updateData] = useState<WistiaVideo[]>([]);
   const [selectedIds, setIds] = useState<number[]>([]);
-  const [dropdownData, filterDropdownData] = useState<WistiaItem[] | []>([]);
+  const [dropdownData, filterDropdownData] = useState<WistiaVideo[]>([]);
   const [loading, updateLoadingStatus] = useState(true);
+  const [end, setEnd] = useState(VIDEOS_PER_PAGE);
   const [error, setError] = useState(null || '');
   sdk.window.startAutoResizer();
 
   // Set inital state based on field values
   useEffect(() => {
+    const fetchInitialVideos = async () => {
+      try {
+        const parameters: any = sdk.parameters.installation;
+        const projectIds = await getProjectIds(
+          parameters.excludedProjects,
+          parameters.apiBearerToken
+        );
+        let videos = await fetchVideos(projectIds, parameters.apiBearerToken);
+        if (videos.length === 0) {
+          setError('There are no videos in your Wistia space');
+        } else {
+          updateData(videos);
+          filterDropdownData(videos);
+        }
+      } catch (error) {
+        if (error instanceof WistiaError) {
+          setError(error.message);
+        }
+      } finally {
+        updateLoadingStatus(false);
+      }
+    };
+
     const fieldValues = sdk.field.getValue();
     setIds(
       fieldValues !== undefined && fieldValues.items.length > 0
-        ? fieldValues.items.map((item: WistiaItem) => item.id)
+        ? fieldValues.items.map((item: WistiaProject) => item.id)
```

Comment:
I know you didn't write this but in line 73, if `fieldValues` is null, it'll pass this check. We should just do 
`fieldValues && fieldValues.items ? etc...`

Reply:
Fixed 👍